certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -421,19 +439,87 @@ def test_reuse_key(context):
     key_path = join(context.workspace, 'privkey-p384.pem')
     csr_path = join(context.workspace, 'csr-p384.der')
     cert_path = join(context.workspace, 'cert-p384.pem')
     chain_path = join(context.workspace, 'chain-p384.pem')
 
-    misc.generate_csr([context.get_domain('ecdsa')], key_path, csr_path, key_type=misc.ECDSA_KEY_TYPE)
-    context.certbot(['auth', '--csr', csr_path, '--cert-path', cert_path, '--chain-path', chain_path])
+    misc.generate_csr(
+        [context.get_domain('ecdsa')],
+        key_path, csr_path,
+        key_type=misc.ECDSA_KEY_TYPE
+    )
+    context.certbot([
+        'auth', '--csr', csr_path, '--cert-path', cert_path,
+        '--chain-path', chain_path,
+    ])
 
     certificate = misc.read_certificate(cert_path)
     assert 'ASN1 OID: secp384r1' in certificate
 
+# - pass no flag or --key-type=rsa and expect a RSA key/cert generated with the default RSA size
+# - pass --key-type=ecdsa and expect an ECDSA key/cert generated with the default ECDSA size
+# - pass wrong --key-type and expect certbot to crash with the expected error message
+# - handle key size control flags (ECDA and RSA) may be added in a modified version of the existing
+# test test_renew_with_changed_private_key_complexity
+
+
+def test_default_key_type(context):
+    """Test default key type is RSA"""
+    certname = context.get_domain('renew')
+    context.certbot([
+        'certonly', '--cert-name', 'newname', '-d', certname
+    ])
+    with open(join(context.config_dir, 'archive/{0}/privkey1.pem').format(certname), 'r') as file:
+        privkey1 = file.read()
+
+    key_data = base64.b64encode('\n'.join(list(map(str.strip, privkey1))[1:-1]))
+    key = load_der_private_key(data=key_data, password=None, backend=default_backend())
+
+    assert isinstance(key, RSAPrivateKey)
+
+
+def test_default_curve_type(context):
+    """test that the curve used when not specifying any is secp256r1"""
+    certname = context.get_domain('renew')
+    context.certbot([
+        'certonly', '--key-type ecdsa', '--cert-name', 'newname', '-d', certname
+    ])
+    key1 = join(context.workspace, 'archive', certname, 'privkey1.pem')
+    assert_elliptic_key(key1, SECP256R1)
+
+
+def test_renew_with_ec_keys(context):
+    """Test proper renew with updated private key complexity."""
+    certname = context.get_domain('renew')
+    context.certbot([
+        'certonly',
+        '-d', certname,
+        '--key-type ecdsa', '--elliptic-curve', 'secp384r1',
+    ])
+
+    key1 = join(context.archive, certname, 'privkey1.pem')
+    assert 200 > os.stat(key1).st_size < 250  # ec keys of 256 bits are ~225 bytes
+    assert_cert_count_for_lineage(context.config_dir, certname, 1)
+
+    context.certbot(['renew'])
+
+    assert_cert_count_for_lineage(context.config_dir, certname, 2)
+    key2 = join(context.config_dir, 'archive', certname, 'privkey2.pem')
+    assert_elliptic_key(key2, SECP384R1)
```

Comment:
I added a new `assert` for this operation, because I used it a few times. I think it's the only way to actually test the type of key used.

Reply:
Makes sense. In this case, could you put the new assertion in the `certbot_integration_tests.certbot_tests.assertions` module? On top of the fact the module logically gather all custom assertions, they are also properly registered as assertions for pytest using `pytest.register_assert_rewrite` call done in `__init__`. It allows pytest to optimize output when these assertions functions raise an `AssertionError`.